rate-limiter/src/main/scala/it/pagopa/interop/commons/ratelimiter/RateLimiterExecutor.scala

```diff
@@ -30,22 +30,22 @@ private[ratelimiter] final case class RateLimiterExecutor(
     ec: ExecutionContext,
     logger: LoggerTakingImplicit[ContextFieldsToLog],
     contexts: Seq[(String, String)]
-  ): Future[Unit] = {
+  ): Future[RateLimitStatus] = {
     val now    = dateTimeSupplier.get
     val result = for {
       // Note: this is not transactional. Potentially N requests can use just 1 token
       bucket <- getBucket(configs.limiterGroup, organizationId, now).recoverWith(clearOnDeserializationError(now))
-      updatedBucket = refillBucket(bucket, now)
-      _ <- useToken(updatedBucket, organizationId)
-    } yield ()
+      refilledBucket = refillBucket(bucket, now)
+      status <- useToken(refilledBucket, organizationId)
+    } yield status
     result
       .recoverWith {
-        case TooManyRequests =>
+        case tmr: TooManyRequests =>
           logger.warn(s"Rate Limit triggered for organization $organizationId")
-          Future.failed(TooManyRequests)
-        case err             =>
+          Future.failed(tmr)
+        case err                  =>
           logger.error(s"Unexpected error during rate limiting for organization $organizationId", err)
-          Future.unit
+          Future.successful(RateLimitStatus(configs.maxRequests, configs.maxRequests, configs.rateInterval))
```

Comment:
Here we are saying that any error except `TooManyRequests` is treated as a success in order to prevent influencing the endpoint logic, is correct?

Reply:
correct